tests/components/elvia/test_config_flow.py

```diff
@@ -0,0 +1,200 @@
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "elvia.meter_value.MeterValue.get_meter_values",
+        return_value={"meteringpoints": []},
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_API_TOKEN: TEST_API_TOKEN,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "no_metering_points"
+
+    assert len(mock_setup_entry.mock_calls) == 0
+
+
+async def test_abort_when_metering_point_id_exist(
+    recorder_mock: Recorder,
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test that we abort when the metering point ID exist."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="1234",
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "elvia.meter_value.MeterValue.get_meter_values",
+        return_value={"meteringpoints": [{"meteringPointId": "1234"}]},
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_API_TOKEN: TEST_API_TOKEN,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "metering_point_id_already_configured"
+
+    assert len(mock_setup_entry.mock_calls) == 0
+
+
+@pytest.mark.parametrize(
+    ("side_effect", "base_error"),
+    (
+        (ElviaError.ElviaException("Boom"), "unknown"),
+        (ElviaError.AuthError("Boom", 403, {}, ""), "invalid_auth"),
+        (ElviaError.ElviaServerException("Boom", 500, {}, ""), "unknown"),
+        (ElviaError.ElviaClientException("Boom"), "unknown"),
+    ),
+)
+async def test_form_exceptions(
+    recorder_mock: Recorder,
+    hass: HomeAssistant,
+    side_effect: Exception,
+    base_error: str,
+) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "elvia.meter_value.MeterValue.get_meter_values",
+        side_effect=side_effect,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_API_TOKEN: TEST_API_TOKEN,
+            },
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": base_error}
```

Comment:
Config flow tests should finish in either Create entry and abort to test they are able to complete